hta/common/trace.py

```diff
@@ -706,7 +706,7 @@ def filter_gpu_kernels_for_one_rank(trace_df: pd.DataFrame) -> pd.DataFrame:
             last_profiler_start = cpu_kernels[cpu_kernels["name"].isin(profiler_steps)][
                 "ts"
             ].max()
-            cpu_kernels = cpu_kernels[cpu_kernels["ts"] < last_profiler_start]
+            cpu_kernels = cpu_kernels[cpu_kernels["ts"] <= last_profiler_start]
```

Comment:
You may change the signature to:
def _filter_irrelevant_gpu_kernels(self, include_last_profiler_step: Optional[bool] = False) -> None:

and then change the code as:
if include_last_profiler_step:
      cpu_kernels = cpu_kernels[cpu_kernels["ts"] <= last_profiler_start]
else:
       cpu_kernels = cpu_kernels[cpu_kernels["ts"] < last_profiler_start]

Reply:
After the change, you can run 
```
make test
```
to verify that all tests run successfully.

Reply:
Thanks for the suggestions! I have added the propagated flag to TraceAnalysis class and also added the corresponding test cases. Related test cases are all passed, but I noticed that one other test case failed, which I believe is unrelated to the changes made here. I have pasted the fail case below:
```
======================================================================
FAIL: test_create_rank_to_trace_dict_with_mixed_dir (tests.test_trace_file.TestTraceFile)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/juntao/github/HolisticTraceAnalysis/tests/test_trace_file.py", line 78, in test_create_rank_to_trace_dict_with_mixed_dir
    self.assertEqual(
AssertionError: Tuples differ: (True, {0: 'tests/data/mixed_files/rank-0.json.gz'}) != (True, {0: 'tests/data/mixed_files/rank_non_gpu.json.gz'})

First differing element 1:
{0: 'tests/data/mixed_files/rank-0.json.gz'}
{0: 'tests/data/mixed_files/rank_non_gpu.json.gz'}

- (True, {0: 'tests/data/mixed_files/rank-0.json.gz'})
?                                        ^^

+ (True, {0: 'tests/data/mixed_files/rank_non_gpu.json.gz'})
?                                        ^^^^^^^^
```